kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/vault/KubeVaultTestKmsFacade.java

```diff
@@ -126,11 +133,50 @@ protected URI getVaultUrl() {
         return URI.create("http://" + vault.getVaultUrl());
     }
 
+    /**
+     * Gets vault version.
+     *
+     * @return the vault version
+     */
+    public String getVaultVersion() {
+        return vault.getVersionInstalled();
+    }
+
     @Override
     public TestKekManager getTestKekManager() {
         return new VaultTestKekManager();
     }
 
+    /**
+     * Is correct version installed boolean.
+     *
+     * @return the boolean
+     */
+    public boolean isCorrectVersionInstalled() {
+        String installedVersion = getVaultVersion();
+        String expectedVersion = VaultTestKmsFacade.HASHICORP_VAULT.split(":")[1];
+
+        return compareTo(installedVersion, expectedVersion) == 0;
+    }
+
+    private int compareTo(String currentVersion, String expectedVersion) {
+        if (expectedVersion == null) {
+            return 1;
+        }
+        String[] currentParts = currentVersion.split("\\.");
+        String[] expectedParts = expectedVersion.split("\\.");
+
+        for (int i = 0; i < expectedParts.length; i++) {
+            int currentPart = i < currentParts.length ? Integer.parseInt(currentParts[i]) : 0;
+            int expectedPart = Integer.parseInt(expectedParts[i]);
+            if (currentPart < expectedPart)
```

Comment:
you can use java.lang.Integer#compare(currentPart, expectedPart) and return != 0